### ISSUE TITLE
Panic if Digital Ocean floating IP not assigned to droplet

### ISSUE THREAD

**Issue by franklinhu** [hashicorp/terraform#4162]
`terraform plan` panics if a floating IP exists, but is not currently assigned to a droplet. 
I've posted a crash log here: https://gist.github.com/franklinhu/5b116f159d491b7550ee

Thanks!

**Comment by stack72** [comment]
Hey @franklinhu, have you got a sample of some config that I can try to reproduce this with? I have some tests that don't see to trigger the same behaviour

**Comment by jen20** [comment]
Relevant part of crash log:

```
2015/12/04 09:42:22 [DEBUG] terraform-provider-digitalocean: goroutine 24 [running]:
2015/12/04 09:42:22 [DEBUG] terraform-provider-digitalocean: github.com/hashicorp/terraform/builtin/providers/digitalocean.resourceDigitalOceanFloatingIpRead(0xc8203f69c0, 0x5eeec0, 0xc8202126c0, 0x0, 0x0)
2015/12/04 09:42:22 [DEBUG] terraform-provider-digitalocean: 	/private/tmp/terraform20151202-21396-jm3im7/terraform-0.6.8/src/github.com/hashicorp/terraform/builtin/providers/digitalocean/resource_digitalocean_floating_ip.go:86 +0x414
2015/12/04 09:42:22 [DEBUG] terraform-provider-digitalocean: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc820141b40, 0xc82051d740, 0x5eeec0, 0xc8202126c0, 0xc8202198f8, 0x0, 0x0)
```

Relevant part of code:

```go
	if _, ok := d.GetOk("droplet_id"); ok {
		log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
		d.Set("region", floatingIp.Droplet.Region.Slug)
	} else {
		d.Set("region", floatingIp.Region.Slug)
	}
```

**Comment by jen20** [comment]
It's failing on the dereference of `floatingIp.Droplet.Region` - we should add a check in there to ensure that is valid.

**Comment by stack72** [comment]
thanks @franklinhu and @jen20 - I just opened a PR that should fix this now #4214
